src/encoding/schema/map.ts

```diff
@@ -424,3 +449,129 @@ export class StringMapSchema extends Schema {
     return map;
   }
 }
+
+function removeRawStringsFromMsgpackValues(
+  value: MsgpackEncodingData
+): MsgpackEncodingData {
+  if (value instanceof RawBinaryString) {
+    return bytesToString(value.rawBinaryValue as Uint8Array);
+  }
+  if (value instanceof Map) {
+    const newMap = new Map<
+      string | number | bigint | Uint8Array,
+      MsgpackEncodingData
+    >();
+    for (const [key, val] of value) {
+      newMap.set(
+        removeRawStringsFromMsgpackValues(key) as
+          | string
+          | number
+          | bigint
+          | Uint8Array,
```

Comment:
what does this `as | string | number` syntax mean?

Reply:
The `as X` syntax is a type cast, and `string | number | bigint | Uint8Array` is the union type of all of these individual types. This cast is needed because the return type of the function is the more general `MsgpackEncodingData`, but map keys can only be a subset of the `MsgpackEncodingData` definition.